src/metabase/api/table.clj

```diff
@@ -79,6 +80,14 @@
         (sync-database/sync-table! updated-table))
       updated-table)))
 
+(defn- format-fields-for-response [resp]
+  (update resp :fields
+          (fn [fields]
+            (map (fn [{:keys [values] :as field}]
+                   (if (seq values)
+                     (update field :values fv/field-values->pairs)
+                     field))
+                 fields))))
```

Comment:
it doesn't really make a huge difference whichever one you think is more readable